instrumentation/network/src/main/java/io/opentelemetry/android/instrumentation/network/NetworkChangeMonitorInstrumentation.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.android.instrumentation.network;
+
+import android.app.Application;
+import androidx.annotation.NonNull;
+import com.google.auto.service.AutoService;
+import io.opentelemetry.android.OpenTelemetryRum;
+import io.opentelemetry.android.instrumentation.AndroidInstrumentation;
+import io.opentelemetry.android.internal.services.ServiceManager;
+import io.opentelemetry.android.internal.services.applifecycle.AppLifecycleService;
+import io.opentelemetry.android.internal.services.network.CurrentNetworkService;
+import io.opentelemetry.android.internal.services.network.data.CurrentNetwork;
+import io.opentelemetry.api.OpenTelemetry;
+import io.opentelemetry.instrumentation.api.instrumenter.AttributesExtractor;
+import io.opentelemetry.instrumentation.api.instrumenter.Instrumenter;
+import java.util.ArrayList;
+import java.util.List;
+
+@AutoService(AndroidInstrumentation.class)
+public final class NetworkChangeMonitorInstrumentation implements AndroidInstrumentation {
```

Comment:
This reads a little clunky now -- I think something just like `NetworkInstrumentation` is sufficient, or even `NetworkChangeInstrumentation`. The "monitor" part of it was supposed to represent something that is looking out for network changes. Maybe a little pedantic, but this instrumentation isn't for the monitor.